packages/data/src/store/reducer.js

```diff
@@ -37,8 +37,37 @@ const isResolved = flowRight( [
 			return nextState;
 		}
 	}
-
 	return state;
 } );
 
-export default isResolved;
+/**
+ * Reducer function returning next state for selector resolution, object form:
+ *
+ *   reducerKey -> selectorName -> EquivalentKeyMap<Array, boolean>
+ *
+ * @param {Object} state   Current state.
+ * @param {Object} action  Dispatched action.
+ *
+ * @return {Object} Next state.
+ */
+const topLevelIsResolved = ( state = {}, action ) => {
```

Comment:
I don't love the name here.  Maybe `rootIsResolved`?

Reply:
I'm trying to understand why this needs to exist at all, where invalidating all resolution for a given store should operate on `isResolved`. The natural form of `isResolved` is a deeply nested object, with the store being the top-level key. In other words, I'd expect the action handler for `INVALIDATE_RESOLUTION_FOR_STORE` to effectively be `omit( state.isResolved, [ action.reducerKey ] )`.

I think it's made a bit difficult by the use of `onSubKey`, but at worst I could still see it being a higher-order reducer which operates after `onSubKey`.

```js
( reducer ) => ( state, action ) =>  (
	action.type === 'INVALIDATE_RESOLUTION_FOR_STORE' ? 
		omit( state, [ action.reducerKey ] ) :
		state
)
```

**Edit:** Okay, I just now realized this is the default reducer function, which made things extra confusing for me. In any case, I think how you have it now is more-or-less the same as suggested, but if my review experience is any evidence, it wasn't immediately clear to me that this wasn't a standalone reducer (standalone in the sense of being another key as part of a `combineReducer` default reducer).

Reply:
It wasn't immediately apparent to me on how to implement this in the existing `flowRight` and `onSubKey` logic which is why I took the path I did.

Reply:
What benefits are there to changing the current logic in this pull?

Reply:
Because (a) it better represents that we're not apply some operation after-the-fact, it's part of the behavior of the `isResolved` reducer and (b) to the point of it appearing misleading as being a separate reducer key by conventions around each function defined in a `reducer.js` with the signature `( state, action )` corresponding to the property of a combined reducer in the default export.

Reply:
So I just did some experimentation and ya I can't just add the new stuff to the existing `isResolved` flowRight behaviour because it's passing along the reducer passed in which assumes its a nested value.  So to condense this I would have to do away with the onSubKey helpers. I'll give a go at that.